snapcraft/internal/meta/snap.py

```diff
@@ -53,6 +52,36 @@
 ]
 
 
+def _get_grade_for_snap(*, base: str, user_set_grade: Optional[str]) -> str:
+    """Return the required grade depending on the base used and user setting."""
+    base_snap_pkg = repo.snaps.SnapPackage(base)
+    is_base_from_non_stable = (
+        "stable" not in base_snap_pkg.get_local_snap_info()["channel"]
+    )
+
+    if not is_base_from_non_stable and user_set_grade is None:
+        logger.warning("'grade' property not specified: defaulting to 'stable'.")
+        grade = "stable"
+    elif not is_base_from_non_stable and user_set_grade is not None:
+        grade = user_set_grade
+    elif is_base_from_non_stable and user_set_grade is None:
+        logger.warning(
+            "'grade' property not specified: defaulting to 'devel' as "
+            "it relies on a base that is not available on the stable channel."
+        )
+        grade = "devel"
+    elif is_base_from_non_stable and user_set_grade == "stable":
+        logger.warning(
```

Comment:
@cjp256 this is the case for when it might be appropriate to fail.

Reply:
What about `validate()`?